Sources/SpeziOnboarding/ConsentConstraint.swift

```diff
@@ -0,0 +1,22 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import PDFKit
+import Spezi
+
+
+/// A Constraint which all `Standard` instances must conform to when using the `OnboardingConsentView`.
+public protocol ConsentConstraint: Standard {
+    /// Adds a new exported consent form represented as `PDFDocument` to the `Standard` conforming to ``ConsentConstraint``.
+    /// 
+    /// - Parameters:
+    ///     - consent: The exported consent form represented as `PDFDocument` that should be added.            
+    ///     - identifier: A 'String' identifying the consent form as specified in `OnboardingConsentView`.                
+    func store(consent: PDFDocument, identifier: String) async throws
```

Comment:
We have had some discussions in the office how to expand the overall setup of the consent document and one idea that came up was the question how we can avoid a breaking change in the future.
One idea that I think we should go for would be to package the consent document in a `ConsentDocumentExport` type that contains the PDF document (e.g. a `pdf` property), the identifier (the type can conform to `Identifiable`). This would allow is to pass this to this function and possibly add more elements to the type without introducing a breaking change as we would for a function signature that would get one more argument.

I would suggest that we pull this change in this transition to the `ConsentConstraint`.

I am also wondering if we should make the PDF property asynchronous (can be a simple wrapper that just exposes a private stored pdf property (e.g. `cachedPDF`) in an asynchronous computed property. While this might sounds strange at the beginning it would allow us to eventually lazily generate the PDF if it is needed by the application. Some apps might not even need the PDF and would just like to get the information from the Consent document and store it in a different way.

CC @AdritRao.

Reply:
Thanks for the suggestion, I think that makes the whole implementation much more flexible! I have updated the code accordingly and introduced the `ConsentDocumentExport` type. I think now it's already much more future-proof :) I also moved the identifiers to a separate type `ConsentDocumentIdentifier` which conforms to `Identifiable`, and have updated the docs accordingly.

I also like the idea of making the pdf property async. I think that's a nice optimization for cases where users do not really need the PDF. I implemented it as you suggested. The `ConsentDocumentExport` can optionally receive a "cachedPDF" property during initialization, and `ConsentDocumentExport.pdf` would then return the cached value. In the `OnboardingConsentView`, this cachedPDF is always set by default when generating the ConsentDocumentExport.

However, in case a cachedPDF is not set, I currently return null for the document, as I have some considerations regarding the "lazy PDF export" as of now. To do that, we may need to store an instance of the `ConsentDocument` in the `ConsentDocumentExport`, to later call `ConsentDocument.export()`. I think this would possibly require making the `ConsentDocument` Sendable, as we might pass the `ConsentDocumentExport` between different actors. As the `ConsentDocument` is a View, I am not sure if that's the best approach. Thus, for now, I did not include this functionality and would like to hear your opinion first. We could also implement this in a future PR, and I could let the async "pdf" property throw something like an "UnimplementedError" for now.